daemon/protocol/daemon.proto

```diff
@@ -8,17 +8,20 @@ syntax = "proto3";
 // protoc -I . daemon.proto --go_out=plugins=grpc:.
 
 
-// Protocol for talking to a Local Scoot Server.
+// Protocol for talking to a Local Scoot Daemon.
 
 package protocol;
 
 service ScootDaemon {
   rpc Echo (EchoRequest) returns (EchoReply) {}
 
-  rpc Run(Command) returns (ProcessStatus) {}
-  rpc Status(StatusQuery) returns (ProcessStatus) {}
+  rpc CreateSnapshot(CreateSnapshotRequest) returns (CreateSnapshotReply) {}
+  rpc CheckoutSnapshot(CheckoutSnapshotRequest) returns (CheckoutSnapshotReply) {}
+  rpc Run(RunRequest) returns (RunReply) {}
+  rpc Poll(PollRequest) returns (PollReply) {}
```

Comment:
This is a unix-style poll, described by PollRequest - <0 for indefinite, 0 for immediate, >0 for timeout.